woocommerce-services.php

```diff
@@ -694,28 +693,35 @@ public function add_tracking_info_to_emails( $order, $sent_to_admin, $plain_text
 
 			// Generate a table row for each label
 			foreach ( $labels as $label ) {
-				$carrier = esc_html( $label['carrier_id'] );
-				$tracking = esc_html( $label['tracking'] );
+				$carrier = strtoupper( $label['carrier_id'] );
+				$tracking = $label['tracking'];
 				$refunded = array_key_exists( 'refund', $label );
 
 				if ( $refunded ) {
 					continue;
 				}
 
+				if ( $plain_text ) {
+					// Should look like '- USPS: 9405536897846173912345' in plain text mode
+					$bullet = '- ';
+					$markup .= $bullet . $carrier . ': ' . $tracking . "\n";
+					continue;
+				}
+
 				$markup .= '<tr>';
-				$markup .= '<td class="td" scope="col">' . strtoupper( $carrier ) . '</td>';
+				$markup .= '<td class="td" scope="col">' . esc_html( $carrier ) . '</td>';
 
 				switch ( $carrier ) {
-					case "fedex":
-						$tracking_url = 'https://www.fedex.com/apps/fedextrack/?action=track&tracknumbers=' . $tracking;
+					case 'fedex':
+						$tracking_url = 'https://www.fedex.com/apps/fedextrack/?action=track&tracknumbers=' . esc_html( $tracking );
 						break;
-					case "usps":
-						$tracking_url = 'https://tools.usps.com/go/TrackConfirmAction.action?tLabels=' . $tracking;
+					case 'usps':
+						$tracking_url = 'https://tools.usps.com/go/TrackConfirmAction.action?tLabels=' . esc_html( $tracking );
```

Comment:
No need to `esc_html()` the tracking number here (or with fedex), as the whole url will be passed to `esc_url()`.  _(see comment below)_